libraries/AP_Scripting/lua_scripts.cpp

```diff
@@ -431,6 +434,23 @@ void lua_scripts::run(void) {
             lua_gc(L, LUA_GCCOLLECT, 0);
 
         } else {
+            if (reload == RELOAD_TRUE) {
+                reload = RELOAD_DELAY;
+                hal.scheduler->delay(10000);
+                continue;
+            } else if (reload == RELOAD_DELAY) {
+                // only try and reload once no matter arming state, no sudden reload after disarming
+                reload = RELOAD_NONE;
+                if (!hal.util->get_soft_armed()) {
+                    load_all_scripts_in_dir(L, SCRIPTING_DIRECTORY);
```

Comment:
The inner logic for this should be moved to a `lua_scripts::load_scripts()` call somewhere rather then requiring us to maintain a mirror copy down here and earlier in the function. Loading logic is something we will probably tweak periodically, so having a single call site to manage it would be good.

Reply:
done